cmd/manager/main.go

```diff
@@ -674,6 +684,53 @@ func chooseAndValidateIPFamily(ipFamilyStr string, ipFamilyDefault corev1.IPFami
 	}
 }
 
+// determineSetDefaultSecurityContext determines what settings we need to use for security context by using the following rules:
+// 1. If the setDefaultSecurityContext is explicitly set to either true, or false, use this value.
+// 2. use openshift detection pulled from kubevirt
+//    https://github.com/kubevirt/kubevirt/blob/f71e9c9615a6c36178169d66814586a93ba515b5/pkg/util/cluster/cluster.go#L21
+//    to determine whether or not we are running within an openshift cluster.  If we determine we are on an openshift cluster,
+//    then default to false, otherwise, return true.
+func determineSetDefaultSecurityContext(setDefaultSecurityContext string, clientset kubernetes.Interface) (bool, error) {
+	if setDefaultSecurityContext == "auto-detect" {
+		return isOpenShift(clientset)
```

Comment:
I think the behaviour here is the opposite of what we want. We do not want to set the `securityContext` if it is OpenShift (because of randomised UIDs and UID 1000 being forbidden by default).

Reply:
Completely agree, and added a suite of unit tests to verify this functionality.